core/src/main/java/org/apache/iceberg/rest/Endpoint.java

```diff
@@ -61,6 +63,19 @@ public class Endpoint {
       Endpoint.create("POST", ResourcePaths.V1_TABLE_REGISTER);
   public static final Endpoint V1_REPORT_METRICS =
       Endpoint.create("POST", ResourcePaths.V1_TABLE_METRICS);
+  public static final Endpoint V1_TABLE_EXISTS = Endpoint.create("HEAD", ResourcePaths.V1_TABLE);
```

Comment:
nit: I would put this entry under `V1_LOAD_TABLE` for grouping similar endpoints.